src/pkrd/reader/gen6.rs

```diff
@@ -82,7 +82,7 @@ pub trait Gen6Reader: Reader {
     }
 
     fn get_party_pkm(&self, slot: PartySlot) -> pkm::Pk6 {
-        let offset = ((slot.value() as usize) * 484) + Self::PARTY_OFFSET;
+        let offset = ((slot.value() as usize - 1) * 484) + Self::PARTY_OFFSET;
```

Comment:
Instead of offsetting this value in multiple places, we should probably adjust `PartySlot`.  If we don't solve the root issue we'll need to offset it everywhere, and it's likely someone will forget at some point.

[These lines](https://github.com/zaksabeast/PokeReader/blob/159a4ddcdfc6025b00e339d10e846392ef6d2862/src/utils/party_slot.rs#L4-L10) should be changed to:
```rs
pub type PartySlot = CircularCounter<0, 5>;

impl fmt::Display for PartySlot {
    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
        write!(f, "{}", self.value() + 1)
    }
}
```

That will ensure all future uses can just be `slot.value()`, and the display will always be correct.